docs/auto-instrumentation-install.md

```diff
@@ -533,10 +529,93 @@ kubectl get certificates
 # splunk-otel-collector-operator-serving-cert   True    splunk-otel-collector-operator-controller-manager-service-cert   5m
 ```
 
-##### Using a Self-Signed Certificate for the Webhook
+#### TLS Certificate Requirement for Kubernetes Operator Webhooks
+
+In Kubernetes, the API server communicates with operator webhook components over HTTPS, which requires a valid TLS certificate that the API server trusts. The operator supports several methods for configuring the required certificate, each with different levels of complexity and security.
+
+---
+
+##### 1. **Using a Self-Signed Certificate Generated by the Chart**
+
+This is the default and simplest method for generating a TLS certificate. It automatically creates a self-signed certificate for the webhook. It is suitable for internal environments or testing purposes but may not be trusted by clients outside your cluster.
+
+**Configuration:**
+- Set `admissionWebhooks.certManager.enabled` to `false` and `admissionWebhooks.autoGenerateCert.enabled` to `true`.
+- Helm generates a self-signed certificate, valid for 10 years, and creates a secret for the webhook.
+- The certificate is automatically recreated on every Helm upgrade.
+
+This is the easiest setup for users and does not require additional configuration.
+
+**Note**: Self-signed certificates are not trusted by default by clients, so this option is generally best for internal or testing scenarios.
+
+---
+
+##### 2. **Using a cert-manager Certificate**
+
+Using `cert-manager` offers more control over certificate management and is more suitable for production environments. However, due to Helm’s install/upgrade order of operations, cert-manager CRDs and certificates cannot be installed within the same Helm operation. To work around this limitation, you can choose one of the following options:
+
+###### Option 1: **Pre-deploy cert-manager**
+
+If `cert-manager` is already deployed in your cluster, you can configure the operator to use it without enabling certificate generation by Helm.
+
+**Configuration:**
+```yaml
+operator:
+  admissionWebhooks:
+    certManager:
+      enabled: true
+    autoGenerateCert:
+      enabled: false
+```
+
+###### Option 2: **Deploy cert-manager and the operator together**
+
+If you need to install `cert-manager` along with the operator, use a Helm post-install or post-upgrade hook to ensure that the certificate is created after cert-manager CRDs are installed.
+
+**Configuration:**
+```yaml
+operator:
+  admissionWebhooks:
+    certManager:
+      enabled: true
+      certificateAnnotations:
+        "helm.sh/hook": post-install,post-upgrade
+        "helm.sh/hook-weight": "1"
+      issuerAnnotations:
+        "helm.sh/hook": post-install,post-upgrade
+        "helm.sh/hook-weight": "1"
+certmanager:
```

Comment:
I am a little confused here. In the chart.yaml, certmanager has been removed from dependency. Are we targeting the removal in this PR?

Reply:
Including these changes was in flux depending a few factors. Last update we are actually going to keep using certmanager as a subchart.

Reply:
sounds good